src/Symfony/Component/HttpKernel/EventListener/DumpListener.php

```diff
@@ -37,8 +39,16 @@ public function __construct(ClonerInterface $cloner, DataDumperInterface $dumper
         $this->dumper = $dumper;
     }
 
-    public function configure()
+    public function configure(GetResponseEvent $event)
     {
+        // if the request is issued by curl, we do not enable HTML dumper;
+        if (false !== strpos($event->getRequest()->headers->get('user-agent'), 'curl/')) {
```

Comment:
I don't understand your comment. It should be `'cli' ===` ? Anyway, it will not work, because here the sapi IS fpm (or mod php) even if the request comes for curl